agent/agent_test.go

```diff
@@ -95,4 +95,78 @@ func TestAgentStartStop(t *testing.T) {
 	}
 
 	assert.NoError(t, agent.Stop(ctx))
+	ready := agent.Ready()
+
+	if ready != nil {
+		t.Fatalf("ready cannot return nil channel")
+	}
+	agent.Start(ctx)
+
+	ready = agent.Ready()
+	if ready != nil {
+		t.Fatalf("ready cannot be nil")
+	}
+
+}
+
+func TestHandleSessionMessage(t *testing.T) {
+	ctx, _ := context.WithTimeout(context.Background(), 5000*time.Millisecond)
+	agent := getAgent(t)
+	peer := api.Peer{"node1", "10.0.0.1"}
+	manager := api.WeightedPeer{&peer, 1.0}
+
+	managers := make([]*api.WeightedPeer, 1)
+	managers[0] = &manager
+	var networkBoostrapKeys = make([]*api.EncryptionKey, 1)
+	networkBoostrapKeys[0] = &api.EncryptionKey{}
+	sessionMessage := api.SessionMessage{"sm1", &api.Node{}, managers, networkBoostrapKeys}
```

Comment:
The entire creation of this message should be a single literal, as I stated above. No need to use 10 lines of code to do this.